### ISSUE TITLE
Sort function doesn't handle transparent materials well

### ISSUE THREAD

**Issue by LagoLunatic** [Astral-C/PyJ3DUltra#4]
Test model:
[model.zip](https://github.com/user-attachments/files/17170568/model.zip)

With PyJ3DUltra's current sort function, the water in this custom model renders like this:
![image](https://github.com/user-attachments/assets/84523d3c-110b-439f-ba58-d700dfb5cc90)

It renders correctly in Jekyll:
![image](https://github.com/user-attachments/assets/dc27db2b-eaeb-4a16-8a21-c7594aef65c8)

If I copy Jekyll's sorting function (`SortFuncTest`) to PyJ3DUltra, it fixes the water:
![image](https://github.com/user-attachments/assets/844f4369-c5e4-451c-acce-c16433bdfeab)
But moving or rotating the camera from its default position at all makes most of the model start rendering as transparent, which didn't happen with the original sort function:
![image](https://github.com/user-attachments/assets/791ed0c0-b585-4e44-9c79-548def6190b3)

I'm not sure what's causing this rendering issue, but it happens for both the example `test/model_viewer.py` in this repo, and for GCFT, but not in Jekyll, so I suspect it's an issue in this repo and not with J3DUltra or GCFT's code.

**Comment by Astral-C** [comment]
Yeah the default sort method isn't very good, should probably make it so user code can define and set the sort function.

**Comment by Astral-C** [comment]
Set sort function to be a bit better and fixed rendering bug in commit 98a66b7, should be good now

**Comment by LagoLunatic** [comment]
Confirmed, thanks!
